File: js/fizzbuzz_array.js
// FizzBuzz JavaScript (JS) - Array

// Create an array
m = [...Array(101).keys()];

// Map the array and print it
console.log(m.map(v).join("\n"));

function v(i) {
  // Create an empty string
  var output = "";
  
  // Divide
  output += check(i, 3, "Fizz");
  output += check(i, 5, "Buzz");
  
  // Check if output is empty
  if (output === "") { output = i; }
  
  return output;
}

function check(i, number, word) {
  if (i % number === 0) { return word; }
  
  return "";
}

File: kotlin/fizzbuzz_array.kt
fun main(args: Array<String>) {
	val max = if (args.size > 0 && args[0].toInt() > 0) args[0].toInt() else 100
	for (i in 1.rangeTo(max)) {
		if (i % 3 == 0 && i % 5 == 0)
			println("FizzBuzz")
		else if (i % 3 == 0)
			println("Fizz")
		else if (i % 5 == 0)
			println("Buzz")
		else
			println(i)
	}
}
